design-proposals/eim-cli.md

```diff
@@ -0,0 +1,254 @@
+- `--deauthorize-only` - optional local flag that only takes effect when `delete` flag is invoked - does not delete host fully but only deauthorizes it
+- `--force-delete` - optional local flag that only takes effect when `delete` flag is invoked - deletes host without deauthorization
+- `--edit-config` `<arguments>` - set of arguments to aid the edition of host
+
+#### List  host information
+
+The list command is responsible for listing all deployed hosts, in tabular format with the amount of information equivalent to what is present in UI: `Name`/`Host Status`/`Serial Number`/`Operating System`/`Site`/`Workload`/`Actions`/`UUID`/`Processor`/`Latest Updates`/`Trusted Compute`   
+> **TODO** this list should probably be scaled back by default and allow for expansion with `-o wide`? Looking for feedback.
+
+- List Registered Host (ie. `emfctl list hosts -o registered`)
+- List Onboarded Host (ie. `emfctl list hosts -o onboarded`)
+- List Provisioned Host (ie. `emfctl list hosts -o provisioned`)
+- List All Host (ie. `emfctl list hosts`) (the default behaviour)
+
+#### Get host information
+
+The get command is responsible for describing the detailed information about a particular host.
+The output is more verbose and includes information from the `list` with addition of the detail captured under `Status Details`/`Resources`/`Specifications`/`I/O Devices`/`OS Profile`/`Host Labels` in UI - likely to be displayed as a mixture human readable sections and tabular? TODO get feedback on how aligned with UI we need this to be could be a table for each subsection.
+
+- Get Host (ie. `emfctl get host myhost`)
+
+#### Bulk import of multiple Hosts
+
+The BIT tool allows for importing a number of hosts through CSV file, on import it registers, onboards and provisions the hosts - a hostname is autogenerated for each added host.
+From the current understanding the tool does not support registration only or onboarding only (ie. the OSProfile is a mandatory field), but it will support handling (onboarding/provisioning) of existing registered hosts if the hosts are added to CSV. With this assumption in mind the proposal is that CLI only uses bulk import for "fully provisioned" hosts (unless/until the BIT supports register only for bulk import).
+
+The bulk import would be invoked with `--import-from-csv` option. Since with bulk import the host names are autogenerated the `host` command/noun would either have to take argument when the flag is provided, or the BIT tool would need to be expanded to accept a name that could be used as a prefix for auto generated host names.
+
+> **TODO** - Look for feedback - I did not give this much thought yet - how do we want to integrate the BIT tool - integrate it into the CLI codebase (then do we need CSV - could it be `yaml` instead) - or integrate to be used with CLI, installed as dependency and CLI calls BIT tool?
+
```

Comment:
I prefer to have BIT integrated in this CLI. Otherwise too many things to remember. We should keep using csv